### ISSUE TITLE
Breaking change in v1.90.0 is not revertable even when downgrade: `unexpected number of substrings in the part name`

### ISSUE THREAD

**Issue by mrceyhun** [VictoriaMetrics/VictoriaMetrics#4336]
### Is your question request related to a specific component?

VictoriaMetrics

### Describe the question in detail

Hi dev team,

fyi @vkuznet @nikodemas @leggerf @brij01

We are running VictoriaMetrics in Kubernetes as a single node. (Actually, not as one instance but for this issue I discarded other unrelated details because we're having this issue in just one specific instance.)

Problem is that:
- The VM instance has been running without any problem with `heads-public-single-node-0-g2ac530eb2` image tag in Kubernetes, at least 30 days.
- We updated the image tag to the latest one and started to get the `unexpected number of substrings in the part name "1760590103B511E0": got 1; want 5` errors.
- We rolled back to the previous image tag of `heads-public-single-node-0-g2ac530eb2`, however, now it gives `cannot parse path to part: unexpected number of substrings in the part name "176058FD6ECDC531": got 1; want 3` errors.

In the CHANGELOG of v1.90.0[1], it says that we should use version greater than `v1.90.0`, but it does not work as I described.

Do you have any suggestion about how to solve this problem?

[1] https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.90.0

I am sharing all the k8s pod error logs in each tag trial:

```
1. ---- (started to use this version and started to get errors) v1.91.0 ----

Image:         victoriametrics/victoria-metrics:v1.91.0
mage ID:      docker.io/victoriametrics/victoria-metrics@sha256:44a35f77867f05ecbadd6aa8026e234a173cee68b0847b9af54b230a0c29d16a

2023-05-22T12:39:56.833Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: victoria-metrics-20230518-212023-tags-v1.91.0-0-g73dfb030d
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -opentsdbHTTPListenAddr=":4242"
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -retentionPeriod="1"
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxConcurrentRequests="32"
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -storageDataPath="/tsdb"
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/app/victoria-metrics/main.go:70	starting VictoriaMetrics at ":8428"...
2023-05-22T12:39:56.833Z	info	VictoriaMetrics/app/vmstorage/main.go:108	opening storage at "/tsdb" with -retentionPeriod=1
2023-05-22T12:39:56.901Z	info	VictoriaMetrics/lib/memory/memory.go:42	limiting caches to 8894656512 bytes, leaving 5929771008 bytes to the OS according to -memory.allowedPercent=60
2023-05-22T12:40:08.493Z	info	VictoriaMetrics/lib/storage/storage.go:874	discarding /tsdb/cache/curr_hour_metric_ids, since it contains outdated hour; got 467217; want 467988
2023-05-22T12:40:08.869Z	info	VictoriaMetrics/lib/storage/storage.go:874	discarding /tsdb/cache/prev_hour_metric_ids, since it contains outdated hour; got 467216; want 467987
2023-05-22T12:40:08.883Z	info	VictoriaMetrics/lib/storage/storage.go:834	discarding /tsdb/cache/next_day_metric_ids, since it contains data for stale date; got 19467; want 19499
2023-05-22T12:40:28.361Z	panic	VictoriaMetrics/lib/storage/part_header.go:142	FATAL: cannot parse metadata from "/tsdb/data/big/2023_05/1760590103B511E0": unexpected number of substrings in the part name "1760590103B511E0": got 1; want 5
panic: FATAL: cannot parse metadata from "/tsdb/data/big/2023_05/1760590103B511E0": unexpected number of substrings in the part name "1760590103B511E0": got 1; want 5

goroutine 1 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0xbb6863, 0x5}, {0xc02bb96780, 0x9f}, 0x2?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:281 +0xa96
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0xbb6863, 0x5}, {0xbd6efb?, 0xc02d2a0ba0?}, {0xc000472d40?, 0x0?, 0xc000472c88?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0x1a5
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partHeader).MustReadMetadata(0xc00002bf20, {0xc00002bef0, 0x27})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/part_header.go:142 +0x353
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mustOpenFilePart({0xc00002bef0?, 0x16?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/part.go:53 +0x6f
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mustOpenParts({0xc015db6a68, 0x16}, {0xc0008d2800, 0x1c, 0xaf64c0?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:1795 +0x42e
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mustOpenPartition({0xc015db6a50?, 0x419860?}, {0xc015db6a68, 0x16}, 0xc0004732f0?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:266 +0x285
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mustOpenPartitions({0xc017d3c080, 0x10}, {0xc02befe050, 0xe}, 0x0?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/table.go:492 +0x26d
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mustOpenTable({0xc000132e00?, 0xb9d700?}, 0xc000320680)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/table.go:109 +0x292
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.MustOpenStorage({0x7fff2f0ae533?, 0xbb6310?}, 0x9fa52400, 0x0, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/storage.go:245 +0xe56
github.com/VictoriaMetrics/VictoriaMetrics/app/vmstorage.Init(0xd64308)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmstorage/main.go:111 +0x4e5
main.main()
	github.com/VictoriaMetrics/VictoriaMetrics/app/victoria-metrics/main.go:74 +0x345


2. ---- (rolled-back to the previous working version but now it gives some errors) heads-public-single-node-0-g2ac530eb2 ----

Image:         victoriametrics/victoria-metrics:heads-public-single-node-0-g2ac530eb2
Image ID:      docker.io/victoriametrics/victoria-metrics@sha256:9cbc9a5d100ab1cb6fe735d0f6614a45f8538addc7785cc3191ca93b64b8d6ba

2023-05-22T12:43:31.626Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: victoria-metrics-20230118-091500-heads-public-single-node-0-g2ac530eb2
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -opentsdbHTTPListenAddr=":4242"
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -retentionPeriod="1"
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxConcurrentRequests="32"
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -storageDataPath="/tsdb"
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/app/victoria-metrics/main.go:58	starting VictoriaMetrics at ":8428"...
2023-05-22T12:43:31.626Z	info	VictoriaMetrics/app/vmstorage/main.go:114	opening storage at "/tsdb" with -retentionPeriod=1
2023-05-22T12:43:31.629Z	info	VictoriaMetrics/lib/memory/memory.go:42	limiting caches to 8894656512 bytes, leaving 5929771008 bytes to the OS according to -memory.allowedPercent=60
2023-05-22T12:43:31.629Z	info	VictoriaMetrics/lib/storage/storage.go:1010	loading MetricName->TSID cache from "/tsdb/cache/metricName_tsid"...
2023-05-22T12:43:33.397Z	info	VictoriaMetrics/lib/storage/storage.go:1015	loaded MetricName->TSID cache from "/tsdb/cache/metricName_tsid" in 1.768 seconds; entriesCount: 4583461; sizeBytes: 3321888768
2023-05-22T12:43:33.397Z	info	VictoriaMetrics/lib/storage/storage.go:1010	loading MetricID->TSID cache from "/tsdb/cache/metricID_tsid"...
2023-05-22T12:43:33.409Z	info	VictoriaMetrics/lib/storage/storage.go:1015	loaded MetricID->TSID cache from "/tsdb/cache/metricID_tsid" in 0.012 seconds; entriesCount: 13739; sizeBytes: 33554432
2023-05-22T12:43:33.409Z	info	VictoriaMetrics/lib/storage/storage.go:1010	loading MetricID->MetricName cache from "/tsdb/cache/metricID_metricName"...
2023-05-22T12:43:33.425Z	info	VictoriaMetrics/lib/storage/storage.go:1015	loaded MetricID->MetricName cache from "/tsdb/cache/metricID_metricName" in 0.015 seconds; entriesCount: 9121; sizeBytes: 33554432
2023-05-22T12:43:33.425Z	info	VictoriaMetrics/lib/storage/storage.go:870	loading curr_hour_metric_ids from "/tsdb/cache/curr_hour_metric_ids"...
2023-05-22T12:43:33.435Z	info	VictoriaMetrics/lib/storage/storage.go:890	discarding /tsdb/cache/curr_hour_metric_ids, since it contains outdated hour; got 467217; want 467988
2023-05-22T12:43:33.435Z	info	VictoriaMetrics/lib/storage/storage.go:870	loading prev_hour_metric_ids from "/tsdb/cache/prev_hour_metric_ids"...
2023-05-22T12:43:33.449Z	info	VictoriaMetrics/lib/storage/storage.go:890	discarding /tsdb/cache/prev_hour_metric_ids, since it contains outdated hour; got 467216; want 467987
2023-05-22T12:43:33.449Z	info	VictoriaMetrics/lib/storage/storage.go:826	loading next_day_metric_ids from "/tsdb/cache/next_day_metric_ids"...
2023-05-22T12:43:33.449Z	info	VictoriaMetrics/lib/storage/storage.go:846	discarding /tsdb/cache/next_day_metric_ids, since it contains data for stale date; got 19467; want 19499
2023-05-22T12:43:33.449Z	info	VictoriaMetrics/lib/mergeset/table.go:319	opening table "/tsdb/indexdb/17579B1FC371422B"...
2023-05-22T12:43:33.463Z	fatal	VictoriaMetrics/app/vmstorage/main.go:119	cannot open a storage at /tsdb with -retentionPeriod=1: cannot open indexdb tables at "/tsdb/indexdb": cannot open curr indexdb table at "/tsdb/indexdb/17579B1FC371422B": cannot open indexDB "/tsdb/indexdb/17579B1FC371422B": cannot open table parts at "/tsdb/indexdb/17579B1FC371422B": cannot open part "/tsdb/indexdb/17579B1FC371422B/176058FD6ECDC531": cannot parse path to part: unexpected number of substrings in the part name "176058FD6ECDC531": got 1; want 3


3. ---- (tried with this version too) v1.89.1 ----
Image:         victoriametrics/victoria-metrics:v1.89.1
Image ID:      docker.io/victoriametrics/victoria-metrics@sha256:d0f855db66fe2c34455e3ac90c6350e82b3c2aa17eaf38e1fe2b4374f04bec51


2023-05-22T12:41:54.849Z	info	VictoriaMetrics/lib/logger/flag.go:12	build version: victoria-metrics-20230313-022110-tags-v1.89.1-0-g388d6ee16
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/lib/logger/flag.go:13	command-line flags
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -opentsdbHTTPListenAddr=":4242"
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -retentionPeriod="1"
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -search.maxConcurrentRequests="32"
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/lib/logger/flag.go:20	  -storageDataPath="/tsdb"
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/app/victoria-metrics/main.go:61	starting VictoriaMetrics at ":8428"...
2023-05-22T12:41:54.849Z	info	VictoriaMetrics/app/vmstorage/main.go:107	opening storage at "/tsdb" with -retentionPeriod=1
2023-05-22T12:41:54.855Z	info	VictoriaMetrics/lib/memory/memory.go:42	limiting caches to 8894656512 bytes, leaving 5929771008 bytes to the OS according to -memory.allowedPercent=60
2023-05-22T12:41:54.855Z	info	VictoriaMetrics/lib/storage/storage.go:1023	loading MetricName->TSID cache from "/tsdb/cache/metricName_tsid"...
2023-05-22T12:41:56.782Z	info	VictoriaMetrics/lib/storage/storage.go:1028	loaded MetricName->TSID cache from "/tsdb/cache/metricName_tsid" in 1.927 seconds; entriesCount: 4583461; sizeBytes: 3321888768
2023-05-22T12:41:56.783Z	info	VictoriaMetrics/lib/storage/storage.go:1023	loading MetricID->TSID cache from "/tsdb/cache/metricID_tsid"...
2023-05-22T12:41:56.801Z	info	VictoriaMetrics/lib/storage/storage.go:1028	loaded MetricID->TSID cache from "/tsdb/cache/metricID_tsid" in 0.018 seconds; entriesCount: 13739; sizeBytes: 33554432
2023-05-22T12:41:56.801Z	info	VictoriaMetrics/lib/storage/storage.go:1023	loading MetricID->MetricName cache from "/tsdb/cache/metricID_metricName"...
2023-05-22T12:41:56.817Z	info	VictoriaMetrics/lib/storage/storage.go:1028	loaded MetricID->MetricName cache from "/tsdb/cache/metricID_metricName" in 0.016 seconds; entriesCount: 9121; sizeBytes: 33554432
2023-05-22T12:41:56.817Z	info	VictoriaMetrics/lib/storage/storage.go:883	loading curr_hour_metric_ids from "/tsdb/cache/curr_hour_metric_ids"...
2023-05-22T12:41:56.826Z	info	VictoriaMetrics/lib/storage/storage.go:903	discarding /tsdb/cache/curr_hour_metric_ids, since it contains outdated hour; got 467217; want 467988
2023-05-22T12:41:56.826Z	info	VictoriaMetrics/lib/storage/storage.go:883	loading prev_hour_metric_ids from "/tsdb/cache/prev_hour_metric_ids"...
2023-05-22T12:41:56.835Z	info	VictoriaMetrics/lib/storage/storage.go:903	discarding /tsdb/cache/prev_hour_metric_ids, since it contains outdated hour; got 467216; want 467987
2023-05-22T12:41:56.835Z	info	VictoriaMetrics/lib/storage/storage.go:839	loading next_day_metric_ids from "/tsdb/cache/next_day_metric_ids"...
2023-05-22T12:41:56.835Z	info	VictoriaMetrics/lib/storage/storage.go:859	discarding /tsdb/cache/next_day_metric_ids, since it contains data for stale date; got 19467; want 19499
2023-05-22T12:41:56.835Z	info	VictoriaMetrics/lib/mergeset/table.go:319	opening table "/tsdb/indexdb/17579B1FC371422B"...
2023-05-22T12:41:56.856Z	fatal	VictoriaMetrics/app/vmstorage/main.go:112	cannot open a storage at /tsdb with -retentionPeriod=1: cannot open indexdb tables at "/tsdb/indexdb": cannot open curr indexdb table at "/tsdb/indexdb/17579B1FC371422B": cannot open indexDB "/tsdb/indexdb/17579B1FC371422B": cannot open table parts at "/tsdb/indexdb/17579B1FC371422B": cannot open part "/tsdb/indexdb/17579B1FC371422B/176058FD6ECDC531": cannot parse path to part: unexpected number of substrings in the part name "176058FD6ECDC531": got 1; want 3
```

### Troubleshooting docs

- [X] General - https://docs.victoriametrics.com/Troubleshooting.html
- [ ] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by Amper** [comment]
Hi @mrceyhun.
Could you show please file tree for dir `/tsdb/data/big/2023_05/1760590103B511E0`?

**Comment by mrceyhun** [comment]
Hi @Amper here it is:
```
$ tree /tsdb/data/big/2023_05/1760590103B511E0
/tsdb/data/big/2023_05/1760590103B511E0
|-- index.bin
|-- metaindex.bin
|-- timestamps.bin
`-- values.bin

0 directories, 4 files
```

**Comment by Amper** [comment]
Something strange happened and for some reason there is no one file you need - `metadata.json`.
Could you check please - is this only directory without `metadata.json` in `/tsdb/data/*/*/*`?

**Comment by mrceyhun** [comment]
Hmm, none of them has `metadata.json`.

**Comment by Amper** [comment]
Sorry for delay.

It is important to say that what is written in the title is not a bug. Backwards-incompatible change was clearly described in [news about release](https://t.me/VictoriaMetrics_en/3720), in [changelog](https://docs.victoriametrics.com/CHANGELOG.html#v1900), and in [release notes](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.90.0). There is no official way to roll back the version. There is only way that you can try at your own risk, that described here: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4113

Nevertheless, you shouldn't have had an error when upgrading to version v1.91.0. I tried to reproduce this issue, but failed. Could you please tell more about how the upgrade process went, were there any other warnings or errors, failures, storage problems, etc...? Judging by the code under regular circumstances such an error should not occur.
Or maybe you have a backup that reproduces the problem?

**Comment by Amper** [comment]
@mrceyhun issue is still catual for you?

**Comment by mrceyhun** [comment]
@Amper Sorry for the late reply. For sure it is our mistake to not properly upgrade the VictoriaMetrics. Also it was having resouce problem and its k8s pod was restarted many times.

I went with the hard way by trusting our multi-instances deployment:
- I deleted the directories named with no underscores (3 part requires 2 underscores AFAIU) from both 'data' and 'indexdb'.
- Then I started VM with previous version before the upgrade.
- Finally, it started with no error. 

Thanks for your help! You may close the issue.

**Comment by f41gh7** [comment]
Hello, an issue must be fixed at attached PR. 

 Workd around for now - deleted bad partitions from log file. It should be safe.

**Comment by f41gh7** [comment]
Was fixed at [v1.91.3](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.91.3) release

**Comment by SaschaP95** [comment]
> > I would like to reopen this Issue. Yesterday I upgraded vmstorage from v1.94.0-cluster to v1.98.0-cluster and got the same error. Hopefully I was able to solve it by downgrade back to v1.94.0
> 
> It's only possible if you upgrade from bugged version ( before v1.91.3). Upgrade from 1.94 to 1.98 shouldn't trigger it.
> 
> It'd be great if you have any reproducible examples.

Sorry it was version v1.93.0-cluster.
I reverted it back and need to check first. When I face the same problem again I will post some logs etc.

**Comment by SaschaP95** [comment]
Hey @f41gh7 

I tried to upgrade it again and i faced the same error.
What I do is to change the version from v1.93.0 to v1.98.0 in the deployment file via vmoperator.

Part of kubectl describe pod vmstorage where you can see that its v1.93.0-cluster
```
Containers:
  vmstorage:
    Container ID:  containerd://4f0aef2fd5dc356f7e58156e61d81138a9599dc81c69b9907e84fe722637ce64
    Image:         XXX/XXX/vmstorage:v1.93.0-cluster
    Image ID:      XXX/XXX/vmstorage@sha256:1a00ae533a8a874068e36ce8289d5f180eb62fa00befb0cc313e2e94a4dace17
    Ports:         8482/TCP, 8400/TCP, 8401/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP
    Args:
      -enableTCP6=true
      -envflag.enable=true
      -httpListenAddr=:8482
      -loggerLevel=INFO
      -pushmetrics.extraLabel=namespace="adns"
      -retentionPeriod=13
      -search.maxUniqueTimeseries=10000000
      -storageDataPath=/vm-data
      -vminsertAddr=:8400
      -vmselectAddr=:8401
```

after the upgrade the same part with v1.98.0-cluster
```
Containers:
  vmstorage:
    Container ID:  containerd://5603e7805078c37aac4a4dd56ee98c7de2f7fcd0df8e4f58ae48932290ecba68
    Image:         XXX/XXX/vmstorage:v1.98.0-cluster
    Image ID:      XXX/XXX/vmstorage@sha256:6f9a71e8600eb2215daf44cd6234d449bc33c0c82f86ee824db9fc0dc2318549
    Ports:         8482/TCP, 8400/TCP, 8401/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP
    Args:
      -enableTCP6=true
      -envflag.enable=true
      -httpListenAddr=:8482
      -loggerLevel=INFO
      -pushmetrics.extraLabel=namespace="adns"
      -retentionPeriod=13
      -search.maxUniqueTimeseries=10000000
      -storageDataPath=/vm-data
      -vminsertAddr=:8400
      -vmselectAddr=:8401
```

and here is the log of the failing vmstorage pod.

```
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: vmstorage-20221219-215809-tags-v1.85.2-cluster-0-g1ed0d07aa
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:13   command-line flags
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -enableTCP6="true"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -envflag.enable="true"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -httpListenAddr=":8482"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -loggerLevel="INFO"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -pushmetrics.extraLabel="\"namespace=\\\"adns\\\"\""
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -retentionPeriod="13"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -search.maxUniqueTimeseries="10000000"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -storageDataPath="/vm-data"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -vminsertAddr=":8400"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/lib/logger/flag.go:20     -vmselectAddr=":8401"
2024-02-27T08:26:06.689Z        info    VictoriaMetrics/app/vmstorage/main.go:94        opening storage at "/vm-data" with -retentionPeriod=13
2024-02-27T08:26:06.692Z        info    VictoriaMetrics/lib/memory/memory.go:42 limiting caches to 943718400 bytes, leaving 629145600 bytes to the OS according to -memory.allowedPercent=60
2024-02-27T08:26:06.692Z        info    VictoriaMetrics/lib/storage/storage.go:1094     loading MetricName->TSID cache from "/vm-data/cache/metricName_tsid"...
2024-02-27T08:26:06.711Z        info    VictoriaMetrics/lib/storage/storage.go:1099     loaded MetricName->TSID cache from "/vm-data/cache/metricName_tsid" in 0.019 seconds; entriesCount: 160; sizeBytes: 9109504
2024-02-27T08:26:06.711Z        info    VictoriaMetrics/lib/storage/storage.go:1094     loading MetricID->TSID cache from "/vm-data/cache/metricID_tsid"...
2024-02-27T08:26:06.715Z        info    VictoriaMetrics/lib/storage/storage.go:1099     loaded MetricID->TSID cache from "/vm-data/cache/metricID_tsid" in 0.005 seconds; entriesCount: 0; sizeBytes: 0
2024-02-27T08:26:06.715Z        info    VictoriaMetrics/lib/storage/storage.go:1094     loading MetricID->MetricName cache from "/vm-data/cache/metricID_metricName"...
2024-02-27T08:26:06.720Z        info    VictoriaMetrics/lib/storage/storage.go:1099     loaded MetricID->MetricName cache from "/vm-data/cache/metricID_metricName" in 0.005 seconds; entriesCount: 0; sizeBytes: 0
2024-02-27T08:26:06.720Z        info    VictoriaMetrics/lib/storage/storage.go:906      loading curr_hour_metric_ids from "/vm-data/cache/curr_hour_metric_ids"...
2024-02-27T08:26:06.721Z        info    VictoriaMetrics/lib/storage/storage.go:976      loaded curr_hour_metric_ids from "/vm-data/cache/curr_hour_metric_ids" in 0.001 seconds; entriesCount: 160; sizeBytes: 2616
2024-02-27T08:26:06.721Z        info    VictoriaMetrics/lib/storage/storage.go:906      loading prev_hour_metric_ids from "/vm-data/cache/prev_hour_metric_ids"...
2024-02-27T08:26:06.722Z        info    VictoriaMetrics/lib/storage/storage.go:976      loaded prev_hour_metric_ids from "/vm-data/cache/prev_hour_metric_ids" in 0.001 seconds; entriesCount: 165; sizeBytes: 2696
2024-02-27T08:26:06.722Z        info    VictoriaMetrics/lib/storage/storage.go:862      loading next_day_metric_ids from "/vm-data/cache/next_day_metric_ids"...
2024-02-27T08:26:06.722Z        info    VictoriaMetrics/lib/storage/storage.go:865      nothing to load from "/vm-data/cache/next_day_metric_ids"
2024-02-27T08:26:06.726Z        info    VictoriaMetrics/lib/mergeset/table.go:308       opening table "/vm-data/indexdb/17B5E715D6B2E2AC"...
2024-02-27T08:26:06.738Z        info    VictoriaMetrics/lib/mergeset/table.go:343       table "/vm-data/indexdb/17B5E715D6B2E2AC" has been opened in 0.012 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2024-02-27T08:26:06.739Z        info    VictoriaMetrics/lib/mergeset/table.go:308       opening table "/vm-data/indexdb/17ACADED38DDFD35"...
2024-02-27T08:26:06.757Z        info    VictoriaMetrics/lib/mergeset/table.go:380       waiting for background workers to stop on "/vm-data/indexdb/17B5E715D6B2E2AC"...
2024-02-27T08:26:06.757Z        info    VictoriaMetrics/lib/mergeset/table.go:383       background workers stopped in 0.000 seconds on "/vm-data/indexdb/17B5E715D6B2E2AC"
2024-02-27T08:26:06.757Z        info    VictoriaMetrics/lib/mergeset/table.go:385       flushing inmemory parts to files on "/vm-data/indexdb/17B5E715D6B2E2AC"...
2024-02-27T08:26:06.757Z        info    VictoriaMetrics/lib/mergeset/table.go:388       inmemory parts have been successfully flushed to files in 0.000 seconds at "/vm-data/indexdb/17B5E715D6B2E2AC"
2024-02-27T08:26:06.757Z        info    VictoriaMetrics/lib/mergeset/table.go:390       waiting for flush callback worker to stop on "/vm-data/indexdb/17B5E715D6B2E2AC"...
2024-02-27T08:26:06.757Z        info    VictoriaMetrics/lib/mergeset/table.go:393       flush callback worker stopped in 0.000 seconds on "/vm-data/indexdb/17B5E715D6B2E2AC"
2024-02-27T08:26:06.759Z        fatal   VictoriaMetrics/app/vmstorage/main.go:98        cannot open a storage at /vm-data with -retentionPeriod=13: cannot open indexdb tables at "/vm-data/indexdb": cannot open prev indexdb table at "/vm-data/indexdb/17ACADED38DDFD35": cannot open indexDB "/vm-data/indexdb/17ACADED38DDFD35": cannot open table parts at "/vm-data/indexdb/17ACADED38DDFD35": cannot open part "/vm-data/indexdb/17ACADED38DDFD35/17B5E715DA4C6E24": cannot parse path to part: unexpected number of substrings in the part name "17B5E715DA4C6E24": got 1; want 3
```

What I see is the build version. I upgraded the cluster multiple times and I think v1.85.2 was the first I started it.

**Comment by SaschaP95** [comment]
I tested it forth and back and there was no chance to upgrade it. So I decided to delete the complete persistentvolumeclaim for one cluster.
After recreation it works as expected and the DB is empty and clear. But when I have a look on the logs again there is the wrong build version.

So vmstorage v1.98.0-cluster clear start and there is still 
```2024-02-27T12:12:23.534Z        info    VictoriaMetrics/lib/logger/flag.go:12   build version: vmstorage-20221219-215809-tags-v1.85.2-cluster-0-g1ed0d07aa```

is it possible that there is a bug in the docker image?
